tasks/Task.cpp

```diff
@@ -196,5 +196,19 @@ void Task::cleanupHook()
     m_driver->close();
     delete m_driver;
     m_driver = 0;
+    for (Mappings::const_iterator it = m_mappings.begin(); it != m_mappings.end(); ++it)
+    {
+        for (size_t i = 0; i < outputports.size(); ++i)
+        {
+            canbus::CanOutputPort const& outputport(outputports[i]);
+            if (it->output->getName() == outputport.ports_name)
```

Comment:
A SEGV can be observed at this line. *it at that time is a zeroed object, so it->output is null. Reason for that probably is usage of m_mappings.erase while keeping the iterator around. (The only iterator that can be taken as guaranteed valid after using erase() is then one returned by erase(). Specifically, the iterator used as parameter becomes invalid.)

Reply:
Also, the delete a few lines down invalidates the it->output object, so the next iteration through outputports, calling getName() is undefined behavior(probably also a SEGV).